front-end/src/components/VolunteerApplication.jsx

```diff
@@ -1,29 +1,89 @@
-import { useState } from "react";
+import { useEffect, useState } from "react";
+import { volunteerRoles } from "../utilities/EventUtilities";
+
 import Button from "react-bootstrap/Button";
 import Form from "react-bootstrap/Form";
 import Modal from "react-bootstrap/Modal";
+import EventDetails from "../pages/EventDetails";
+import { resolvePath } from "react-router-dom";
 
+//name, email, phone, best time to contact, have you volunteered with this organization/person before, why would you like to volunteer for this event, what skills/experience do you have? which role are you interested in?
 
+function VolunteerApplication({ show, handleClose, eventID }) {
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [phoneNum, setPhoneNum] = useState("");
+  const [contactTime, setContactTime] = useState("");
+  const [repeatVolunteer, setRepeatVolunteer] = useState("no");
+  const [whyWantToVol, setWhyWantToVol] = useState("");
+  const [skillsDescript, setSkillsDescript] = useState("");
+  const [rolesSelected, setRolesSelected] = useState([]);
 
-//name, email, phone, best time to contact, have you volunteered with this organization/person before, why would you like to volunteer for this event, what skills/experience do you have? which role are you interested in?
+  const [roles, setRoles] = useState([]);
 
+  //setting roles array by making api call for specific event
+  const getVolunteerRoles = async () => {
+    let rolesArr = await volunteerRoles(eventID); //{id, assigned_volunteers, role, num_volunteers_needed, event}
+    setRoles(rolesArr);
+  };
 
+  useEffect(() => {
+    getVolunteerRoles();
+  }, [eventID]);
 
-function VolunteerApplication({show, handleClose}) {
-  
+  //TODO - incorporate actual application API call & uncomment submit code
+  const handleSubmit = async () => {
+    handleClose();
+    // try{
+    //   await postApplication(
+    //     name,
+    //     email,
+    //     phoneNum,
+    //     contactTime,
+    //     repeatVolunteer,
+    //     whyWantToVol,
+    //     skillsDescript,
+    //    rolesSelected
+    //   );
+    // alert("Your application has been submitted!");
+    // window.location.reload();
+    // } catch (error) {
+    //   console.log('Failed application submission', error);
+    // }
+  };
+
+  //handles checkbox selecting/deselecting, updates rolesSelected state accordingly
+  function handleClickRole(roleID) {
+
+    // find the role instance that matches the provided roleID
+    const roleInstance = roles.find((r) => r.id === roleID);
```

Comment:
Is this just leftover from when you were pulling the whole instance?

Reply:
ahh yes, i'll update it now.